Explorer/Assets/Scripts/ECS/StreamableLoading/Tests/LoadSystemBaseShould.cs

```diff
@@ -0,0 +1,148 @@
+
+        protected abstract TSystem CreateSystem();
+
+        private AssetPromise<TAsset, TIntention> promise;
+
+        protected IStreamableCache<TAsset, TIntention> cache;
+
+        [SetUp]
+        public void BaseSetUp()
+        {
+            cache = Substitute.For<IStreamableCache<TAsset, TIntention>>();
+            system = CreateSystem();
+            system.Initialize();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            promise.LoadingIntention.CommonArguments.cancellationTokenSource?.Cancel();
+        }
+
+        [Test]
+        public async Task ConcludeSuccess()
+        {
+            promise = AssetPromise<TAsset, TIntention>.Create(world, CreateSuccessIntention());
+
+            // Launch the flow
+            system.Update(0);
+
+            StreamableLoadingResult<TAsset> result = await promise.ToUniTask(world, cancellationToken: promise.LoadingIntention.CommonArguments.CancellationToken);
+
+            Assert.That(result.Succeeded, Is.True);
+            Assert.That(result.Asset, Is.Not.Null);
+
+            AssertSuccess(result.Asset);
+        }
+
+        [Test]
+        public async Task ConcludeExceptionOnParseFail()
+        {
+            promise = AssetPromise<TAsset, TIntention>.Create(world, CreateWrongTypeIntention());
+
+            // Launch the flow
+            system.Update(0);
+
+            StreamableLoadingResult<TAsset> result = await promise.ToUniTask(world, cancellationToken: promise.LoadingIntention.CommonArguments.CancellationToken);
+
+            Assert.That(result.Succeeded, Is.False);
+            Assert.IsNotNull(result.Exception);
+        }
+
+        [Test]
+        public async Task ConcludeFailIfNotFound()
+        {
+            TIntention intent = CreateNotFoundIntention();
+            intent.SetAttempts(1);
+
+            promise = AssetPromise<TAsset, TIntention>.Create(world, intent);
+
+            // Launch the flow
+            system.Update(0);
+
+            StreamableLoadingResult<TAsset> result = await promise.ToUniTask(world, cancellationToken: promise.LoadingIntention.CommonArguments.CancellationToken);
+
+            Assert.IsFalse(result.Succeeded);
+            Assert.IsNotNull(result.Exception);
+        }
+
+        [Test]
+        public async Task RemoveCurrentSourceFromPermittedSources()
+        {
+            TIntention intent = CreateSuccessIntention();
+            intent.SetSources(AssetSource.EMBEDDED, AssetSource.EMBEDDED);
+
+            promise = AssetPromise<TAsset, TIntention>.Create(world, CreateSuccessIntention());
+
+            // Launch the flow
+            system.Update(0);
+
+            Assert.AreEqual(AssetSource.NONE, world.Get<TIntention>(promise.Entity).CommonArguments.PermittedSources);
+            await promise.ToUniTask(world, cancellationToken: promise.LoadingIntention.CommonArguments.CancellationToken);
+        }
+
+        [Test]
+        public async Task GetAssetFromCache()
```

Comment:
Could we add a test here to cover the CreateLinkedTokenSource code? I was struggling to follow the flow so it would be useful to know its covered by tests.